### ISSUE TITLE
Feature Request: Text Generation

### ISSUE THREAD

**Issue by byronetodd** [Le0Michine/FusionGridfinityGenerator#72]
Hi,

Awesome addon! It would be that much better with the ability to add text to the label tab like so:
![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/74696326/92dd4ec1-a25a-418b-b76c-1cfb70536aa1)

**Comment by pcout** [comment]
:+1:

**Comment by muppy** [comment]
I agree, that would be great!

**Comment by t-makaro** [comment]
You can add text with a sketch. You should even be able to use the existing geometry to consistently align everything.
![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/18018386/213c8afd-390f-4a97-858b-8983fd0da91c)

**Comment by muppy** [comment]
> You can add text with a sketch. You should even be able to use the existing geometry to consistently align everything. ![image](https://private-user-images.githubusercontent.com/18018386/310660442-213c8afd-390f-4a97-858b-8983fd0da91c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5NDkzMjcsIm5iZiI6MTcxOTk0OTAyNywicGF0aCI6Ii8xODAxODM4Ni8zMTA2NjA0NDItMjEzYzhhZmQtMzkwZi00YTk3LTg1OGItODk4M2ZkMGRhOTFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDE5MzcwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlMDZkZjE0MDE0ZTJkMTFhNzQwOTc4ZmFlOGM4Mjc1MGE1ZWE2YjUwZDE0YmU5Y2RhOGIyMDg4ZDVlMWFiNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WkiGE_x-8nDawoifHWX8HFJj_TWg_ZdGz4cgjGs5XZk)

This solution is very uncomfortable. Is there any progress on this request ?

**Comment by byronetodd** [comment]
> You can add text with a sketch. You should even be able to use the existing geometry to consistently align everything. ![image](https://private-user-images.githubusercontent.com/18018386/310660442-213c8afd-390f-4a97-858b-8983fd0da91c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2ODU3ODIsIm5iZiI6MTczMzY4NTQ4MiwicGF0aCI6Ii8xODAxODM4Ni8zMTA2NjA0NDItMjEzYzhhZmQtMzkwZi00YTk3LTg1OGItODk4M2ZkMGRhOTFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA4VDE5MTgwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjZTc2NjJjMjg1OTRkNzE3ZThjODEwOTBhNjhhNjc4MTNhNjk4YTc2MGI3MjA0YjAzYmMxZGVlMGI2MGUzMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LefQHYdTkmygXmnN4TvAm8qleyNMwoI_nRddPwUeFvM)

Yes, that’s exactly what I have been doing. It would be much easier if it was integrated into the plug in though.

**Comment by Le0Michine** [comment]
not sure how easy that would be to implement, fusion doesn't allow to use parameters for text, so in case it doesn't fit right it maybe hard to adjust. But that's an interesting thing to look into. Might be able to add some naive implementation, though it's likely to break if text is too long

**Comment by byronetodd** [comment]
> not sure how easy that would be to implement, fusion doesn't allow to use parameters for text, so in case it doesn't fit right it maybe hard to adjust. But that's an interesting thing to look into. Might be able to add some naive implementation, though it's likely to break if text is too long

There is actually an add in for parametric text that I started to use. Not sure if it could be integrated into the project easily.

https://github.com/thomasa88/ParametricText
